File: exercises/src/main/java/BackpressureBase.java
import reactor.core.publisher.Flux;
import reactor.test.publisher.TestPublisher;

import java.util.UUID;

/**
 * @author Stefan Dragisic
 */
public class BackpressureBase {

    TestPublisher<String> pub1 = TestPublisher.create();

    public Flux<String> messageStream1() {
        return pub1.flux();
    }

    TestPublisher<String> pub2 = TestPublisher.create();

    public Flux<String> messageStream2() {
        return pub2.flux();
    }

    TestPublisher<String> pub3 = TestPublisher.createNoncompliant(TestPublisher.Violation.REQUEST_OVERFLOW);

    public Flux<String> messageStream3() {
        return pub3.flux();
    }

    TestPublisher<String> pub4 = TestPublisher.createNoncompliant(TestPublisher.Violation.REQUEST_OVERFLOW);

    public Flux<String> messageStream4() {
        return pub4.flux();
    }

    public Flux<String> remoteMessageProducer() {
        return Flux.generate(s -> s.next("MESSAGE#" + UUID.randomUUID()));
    }
}

File: exercises/src/main/kotlin/BackpressureBase.kt
import reactor.core.publisher.Flux
import reactor.test.publisher.TestPublisher
import java.util.UUID

/**
 * @author Stefan Dragisic
 */
class BackpressureBase {
    var pub1: TestPublisher<String> = TestPublisher.create()
    fun messageStream1(): Flux<String> {
        return pub1.flux()
    }

    var pub2: TestPublisher<String> = TestPublisher.create()
    fun messageStream2(): Flux<String> {
        return pub2.flux()
    }

    var pub3: TestPublisher<String> = TestPublisher.createNoncompliant(TestPublisher.Violation.REQUEST_OVERFLOW)
    fun messageStream3(): Flux<String> {
        return pub3.flux()
    }

    var pub4: TestPublisher<String> = TestPublisher.createNoncompliant(TestPublisher.Violation.REQUEST_OVERFLOW)
    fun messageStream4(): Flux<String> {
        return pub4.flux()
    }

    fun remoteMessageProducer(): Flux<String> {
        return Flux.generate { s -> s.next("MESSAGE#" + UUID.randomUUID()) }
    }
}